modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -121,3 +121,81 @@ func (suite *KeeperTestSuite) TestChanUpgradeInit() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestRestoreChannel() {
+	var (
+		path           *ibctesting.Path
+		channelUpgrade types.Channel
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"succeeds when restore channel is set",
+			func() {},
+			true,
+		},
+		{
+			name: "fails when no restore channel is present",
+			malleate: func() {
+				// remove the restore channel
+				path.EndpointA.Chain.GetSimApp().IBCKeeper.ChannelKeeper.DeleteUpgradeRestoreChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			},
+			expPass: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			originalChannel := path.EndpointA.GetChannel()
+
+			chanCap, _ := suite.chainA.GetSimApp().GetScopedIBCKeeper().GetCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
+			channelUpgrade = types.NewChannel(types.INITUPGRADE, types.UNORDERED, types.NewCounterparty(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID), []string{path.EndpointA.ConnectionID}, fmt.Sprintf("%s-v2", mock.Version))
+
+			tc.malleate()
+
+			sequence, _, err := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeInit(
+				suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID,
+				chanCap, channelUpgrade, path.EndpointB.Chain.GetTimeoutHeight(), 0,
+			)
+
+			suite.Require().NoError(err)
+
+			// update the channel to have some other state.
+			modifiedChannel := originalChannel
+			modifiedChannel.Version = "different-version"
+			modifiedChannel.ConnectionHops = []string{"different-connection-id"}
+			path.EndpointA.Chain.GetSimApp().IBCKeeper.ChannelKeeper.SetChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, modifiedChannel)
```

Comment:
Why do you have this code block?

Reply:
after discussion on slack: It would make more sense to actually test `ibckeeper. ChanUpgradeInit` as we would be better able to test the full flow.